### PR TITLE
backport action: run only when the label starts with 'backport'

### PR DESCRIPTION
**Pull request by domenkozar**
I haven't tested this, but it looks like it should work.

### CODE REVIEW

**Review by blaggacao — CHANGES_REQUESTED** [review]
The [zeebe docs suggest](https://github.com/zeebe-io/backport-action) that the label selector should be done differently.

```
          ​#​ Optional​
          ​#​ Regex pattern to match github labels​
          ​#​ Must contain a capture group for target branchname​
          ​#​ label_pattern: ^backport
```

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ on:
 jobs:
   backport:
     name: Backport Pull Request
-    if: github.repository_owner == 'NixOS' && github.event.pull_request.merged == true
+    if: github.repository_owner == 'NixOS' && github.event.pull_request.merged == true && (github.event_name != 'labeled' || startsWith('backport', github.event.label.name))
```
Comment (dotlambda) [inline .github/workflows/backport.yml:8]: Shouldn't it also be triggered by a merge event?

Comment (domenkozar) [reply .github/workflows/backport.yml:8]: `github.event_name != 'labeled'` will be `true` then

**Comment by domenkozar** [comment]
> The [zeebe docs suggest](https://github.com/zeebe-io/backport-action) that the label selector should be done differently.
> 
> ```
>           ​#​ Optional​
>           ​#​ Regex pattern to match github labels​
>           ​#​ Must contain a capture group for target branchname​
>           ​#​ label_pattern: ^backport
> ```

That's orthogonal, it allows us to define what label is interpreted as valid backport label. The linked issue will still reproduce.

File: .github/workflows/backport.yml (hunk 1 above)
Comment (dotlambda) [reply .github/workflows/backport.yml:8]: Oops, I misread it.

**Review by blaggacao — APPROVED** [review]
LGTM

**Review by Ma27 — APPROVED** [review]

**Comment by zowoq** [comment]
@domenkozar 

Seems this label condition stopped working at some point, now it is running for every merged PR.

e.g.

https://github.com/NixOS/nixpkgs/pull/182370
https://github.com/NixOS/nixpkgs/runs/7498116953

@winterqt You may be interested in taking a look at it?

**Pull request merged**